eva/udfs/yolo_object_detector.py

```diff
@@ -0,0 +1,199 @@
+        self.model = torch.hub.load("ultralytics/yolov5", "yolov5m")
+
+    @property
+    def input_format(self) -> FrameInfo:
+        return FrameInfo(-1, -1, 3, ColorSpace.RGB)
+
+    @property
+    def labels(self) -> List[str]:
+        return [
+            "__background__",
+            "person",
+            "bicycle",
+            "car",
+            "motorcycle",
+            "airplane",
+            "bus",
+            "train",
+            "truck",
+            "boat",
+            "traffic light",
+            "fire hydrant",
+            "N/A",
+            "stop sign",
+            "parking meter",
+            "bench",
+            "bird",
+            "cat",
+            "dog",
+            "horse",
+            "sheep",
+            "cow",
+            "elephant",
+            "bear",
+            "zebra",
+            "giraffe",
+            "N/A",
+            "backpack",
+            "umbrella",
+            "N/A",
+            "N/A",
+            "handbag",
+            "tie",
+            "suitcase",
+            "frisbee",
+            "skis",
+            "snowboard",
+            "sports ball",
+            "kite",
+            "baseball bat",
+            "baseball glove",
+            "skateboard",
+            "surfboard",
+            "tennis racket",
+            "bottle",
+            "N/A",
+            "wine glass",
+            "cup",
+            "fork",
+            "knife",
+            "spoon",
+            "bowl",
+            "banana",
+            "apple",
+            "sandwich",
+            "orange",
+            "broccoli",
+            "carrot",
+            "hot dog",
+            "pizza",
+            "donut",
+            "cake",
+            "chair",
+            "couch",
+            "potted plant",
+            "bed",
+            "N/A",
+            "dining table",
+            "N/A",
+            "N/A",
+            "toilet",
+            "N/A",
+            "tv",
+            "laptop",
+            "mouse",
+            "remote",
+            "keyboard",
+            "cell phone",
+            "microwave",
+            "oven",
+            "toaster",
+            "sink",
+            "refrigerator",
+            "N/A",
+            "book",
+            "clock",
+            "vase",
+            "scissors",
+            "teddy bear",
+            "hair drier",
+            "toothbrush",
+        ]
+
+    def forward(self, frames: Tensor) -> pd.DataFrame:
+        """
+        Performs predictions on input frames
+        Arguments:
+            frames (np.ndarray): Frames on which predictions need
+            to be performed
+
+        Returns:
+            tuple containing predicted_classes (List[List[str]]),
+            predicted_boxes (List[List[BoundingBox]]),
+            predicted_scores (List[List[float]])
+
+        """
+        # Stacking all frames, and changing to numpy
+        # because of yolov5 error with Tensors
+
+        for i in range(frames.shape[0]):
```

Comment:
Why aren't we doing batched inference? https://github.com/ultralytics/yolov5/issues/1806

Reply:
Yolov5 requires the H and W of image to be a multiple of 32, however it is not guaranteed for all images. This causes `RuntimeError`.
However, when given images in `numpy`/`cv2 image` format it auto scales them to the proper format.
It also does not accept batch input in `numpy` format